internal/local/structure_v2.go

```diff
@@ -331,3 +328,228 @@ func parseSync(rootDir string) (*SyncStructure, error) {
+	temp := &struct {
+		ConfigVersion         realm.AppConfigVersion `json:"config_version"`
+		ID                    string                 `json:"app_id,omitempty"`
+		Name                  string                 `json:"name,omitempty"`
+		Location              realm.Location         `json:"location,omitempty"`
+		DeploymentModel       realm.DeploymentModel  `json:"deployment_model,omitempty"`
+		Environment           string                 `json:"environment,omitempty"`
+		AllowedRequestOrigins []string               `json:"allowed_request_origins,omitempty"`
+	}{
+		ConfigVersion:         a.ConfigVersion(),
+		ID:                    a.ID(),
+		Name:                  a.Name(),
+		Location:              a.Location(),
+		DeploymentModel:       a.DeploymentModel(),
+		Environment:           a.Environment,
+		AllowedRequestOrigins: a.AllowedRequestOrigins,
+	}
+	return json.Marshal(temp)
+}
+
+// WriteData will write the local Realm app data to disk
+func (a AppDataV2) WriteData(rootDir string) error {
+	if err := writeSecrets(rootDir, a.Secrets); err != nil {
+		return err
+	}
+	if err := writeEnvironments(rootDir, a.Environments); err != nil {
+		return err
+	}
+	if err := writeValues(rootDir, a.Values); err != nil {
+		return err
+	}
+	if err := writeGraphQL(rootDir, a.GraphQL); err != nil {
+		return err
+	}
+	if err := writeServices(rootDir, a.Services); err != nil {
+		return err
+	}
+	if err := writeFunctionsV2(rootDir, a.Functions); err != nil {
+		return err
+	}
+	if err := writeAuth(rootDir, a.Auth); err != nil {
+		return err
+	}
+	if err := writeSync(rootDir, a.Sync); err != nil {
+		return err
+	}
+	if err := writeDataSources(rootDir, a.DataSources); err != nil {
+		return err
+	}
+	if err := writeHTTPEndpoints(rootDir, a.HTTPEndpoints); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeFunctionsV2(rootDir string, functions *FunctionsStructure) error {
+	if functions != nil {
+		for _, function := range functions.Configs {
+			nameFunc := function["name"].(string)
+			dataFunc, err := MarshalJSON(function)
+			if err != nil {
+				return err
+			}
+			dataFunc = append(dataFunc, "\n"...)
+			err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunc, FileConfig.String()), 0666, bytes.NewReader(dataFunc))
+			if err != nil {
+				return err
+			}
+			dataSrc := functions.Sources[nameFunc]
+			err = WriteFile(filepath.Join(rootDir, NameFunctions, nameFunc, FileSource.String()), 0666, bytes.NewReader([]byte(dataSrc)))
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		err := WriteFile(filepath.Join(rootDir, NameFunctions, FileConfig.String()), 0666, bytes.NewReader([]byte("[]\n")))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func writeAuth(rootDir string, auth *AuthStructure) error {
+	if auth != nil && auth.Providers != nil {
+		dataProvider, err := MarshalJSON(auth.Providers)
+		if err != nil {
+			return err
+		}
+		dataProvider = append(dataProvider, "\n"...)
```

Comment:
question, do we not need the newline for other outputs from `MarshalJSON`?  i don't see it _everywhere_, is why i ask.

if we do need it, lets just add that to the `MarshalJSON` implementation.  alternatively, i know `json.NewDecoder(...).Decode(..)` will automatically append that newline, so maybe we just switch to using that?